Test failures:

https://github.com/apache/airflow/actions/runs/15992030386/job/45107520283?pr=52618#step:6:1117
```
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_msgraph.py::TestMSGraphAsyncOperator::test_execute_with_old_result_processor_signature - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_msgraph.py::TestMSGraphAsyncOperator::test_execute_with_new_result_processor_signature - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_msgraph.py::TestMSGraphAsyncOperator::test_execute_with_old_paginate_function_signature - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_powerbi.py::TestPowerBIDatasetRefreshOperator::test_powerbi_operator_async_get_refresh_status_success - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_powerbi.py::TestPowerBIDatasetRefreshOperator::test_powerbi_operator_async_execute_complete_success - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_powerbi.py::TestPowerBIDatasetRefreshOperator::test_powerbi_operator_async_execute_complete_fail - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_powerbi.py::TestPowerBIDatasetRefreshOperator::test_powerbi_operator_refresh_fail - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_powerbi_list.py::TestPowerBIDatasetListOperator::test_powerbi_operator_async_execute_complete_success - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_powerbi_list.py::TestPowerBIDatasetListOperator::test_powerbi_operator_async_execute_complete_fail - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_powerbi_list.py::TestPowerBIDatasetListOperator::test_powerbi_operator_dataset_list_fail - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_powerbi_list.py::TestPowerBIWorkspaceListOperator::test_powerbi_operator_async_execute_complete_success - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_powerbi_list.py::TestPowerBIWorkspaceListOperator::test_powerbi_operator_async_execute_complete_fail - KeyError: 'task_instance'
FAILED providers/microsoft/azure/tests/unit/microsoft/azure/operators/test_powerbi_list.py::TestPowerBIWorkspaceListOperator::test_powerbi_operator_workspace_list_fail - KeyError: 'task_instance'
```

--- a/providers/microsoft/azure/src/airflow/providers/microsoft/azure/operators/msgraph.py
+++ b/providers/microsoft/azure/src/airflow/providers/microsoft/azure/operators/msgraph.py
@@ -307,7 +307,7 @@ class MSGraphAsyncOperator(BaseOperator):
                 self.key,
                 value,
             )
-            context["task_instance"].xcom_push(key=self.key, value=value)
+            context["ti"].xcom_push(key=self.key, value=value)
 
     @staticmethod
     def paginate(

--- a/providers/microsoft/azure/src/airflow/providers/microsoft/azure/operators/powerbi.py
+++ b/providers/microsoft/azure/src/airflow/providers/microsoft/azure/operators/powerbi.py
@@ -141,7 +141,7 @@ class PowerBIDatasetRefreshOperator(BaseOperator):
             dataset_refresh_id = event["dataset_refresh_id"]
 
         if dataset_refresh_id:
-            context["task_instance"].xcom_push(
+            context["ti"].xcom_push(
                 key=f"{self.task_id}.powerbi_dataset_refresh_Id",
                 value=dataset_refresh_id,
             )
@@ -167,7 +167,7 @@ class PowerBIDatasetRefreshOperator(BaseOperator):
         Relies on trigger to throw an exception, otherwise it assumes execution was successful.
         """
         if event:
-            context["task_instance"].xcom_push(
+            context["ti"].xcom_push(
                 key=f"{self.task_id}.powerbi_dataset_refresh_status",
                 value=event["dataset_refresh_status"],
             )
@@ -233,7 +233,7 @@ class PowerBIWorkspaceListOperator(BaseOperator):
         Relies on trigger to throw an exception, otherwise it assumes execution was successful.
         """
         if event:
-            context["task_instance"].xcom_push(
+            context["ti"].xcom_push(
                 key=f"{self.task_id}.powerbi_workspace_ids",
                 value=event["workspace_ids"],
             )
@@ -303,7 +303,7 @@ class PowerBIDatasetListOperator(BaseOperator):
         Relies on trigger to throw an exception, otherwise it assumes execution was successful.
         """
         if event:
-            context["task_instance"].xcom_push(
+            context["ti"].xcom_push(
                 key=f"{self.task_id}.powerbi_dataset_ids",
                 value=event["dataset_ids"],
             )

--- a/providers/microsoft/azure/src/airflow/providers/microsoft/azure/transfers/sftp_to_wasb.py
+++ b/providers/microsoft/azure/src/airflow/providers/microsoft/azure/transfers/sftp_to_wasb.py
@@ -32,6 +32,7 @@ if TYPE_CHECKING:
 from airflow.exceptions import AirflowException
 from airflow.models import BaseOperator
 from airflow.providers.microsoft.azure.hooks.wasb import WasbHook
+from airflow.providers.microsoft.azure.version_compat import AIRFLOW_V_3_0_PLUS
 from airflow.providers.sftp.hooks.sftp import SFTPHook
 
 WILDCARD = "*"
@@ -98,6 +99,8 @@ class SFTPToWasbOperator(BaseOperator):
         self.create_container = create_container
 
     def dry_run(self) -> None:
+        if not AIRFLOW_V_3_0_PLUS:
+            raise NotImplementedError("Not implemented for Airflow 3.")
         super().dry_run()
         sftp_files: list[SftpFile] = self.get_sftp_files_map()
         for file in sftp_files:

